modules/abstract-eth/src/lib/utils.ts

```diff
@@ -672,37 +676,98 @@ export function getV1WalletInitializationData(walletOwners: string[], salt: stri
 }
 
 /**
- * Returns the create address method calling data for v1 wallets
+ * Returns the create address method calling data for v1, v2, v4 forwarders
  *
  * @param {string} baseAddress - The address of the wallet contract
  * @param {string} salt - The salt for address initialization transactions
+ * @param {string} feeAddress - The fee address for the enterprise
  * @returns {string} - the createForwarder method encoded
  */
-export function getV1AddressInitializationData(baseAddress: string, salt: string): string {
+export function getV1AddressInitializationData(baseAddress: string, salt: string, feeAddress?: string): string {
   const saltBuffer = setLengthLeft(toBuffer(salt), 32);
-  const params = [baseAddress, saltBuffer];
-  const method = EthereumAbi.methodID('createForwarder', createV1ForwarderTypes);
-  const args = EthereumAbi.rawEncode(createV1ForwarderTypes, params);
+  const { createForwarderParams, createForwarderTypes } = getCreateForwarderParamsAndTypes(
+    baseAddress,
+    saltBuffer,
+    feeAddress
+  );
+
+  const method = EthereumAbi.methodID('createForwarder', createForwarderTypes);
+  const args = EthereumAbi.rawEncode(createForwarderTypes, createForwarderParams);
   return addHexPrefix(Buffer.concat([method, args]).toString('hex'));
 }
 
+/**
+ * Returns the create address method calling data for all forwarder versions
+ *
+ * @param {number} forwarderVersion - The version of the forwarder to create
+ * @param {string} baseAddress - The address of the wallet contract
+ * @param {string} salt - The salt for address initialization transactions
+ * @param {string} feeAddress - The fee address for the enterprise
+ * @returns {string} - the createForwarder method encoded
+ *
+ */
+export function getAddressInitDataAllForwarderVersions(
+  forwarderVersion: number,
+  baseAddress: string,
+  salt: string,
+  feeAddress?: string
+): string {
+  if (forwarderVersion === defaultForwarderVersion) {
```

Comment:
Whats the `defaultForwarderVersion` now ? how is it in sync with the one in wp?
should we have it per coin?

Reply:
Actually this defaultForwarderVersion is not same as WP as it is not needed here ideally. In WP defaultForwarderVersion is per coin basis. This was mainly added when we were onboarding polygon and we had to add logic related to v1/v2 forwarder init. So we had referred to v0 as defaultForwarderVersion

Reply:
So I have try to create a forwarder using bitgojs without giving any version, it will default to 0 ?

Reply:
I tried by calling sdk createAddress method without giving a forwarderVersion, it seems like it is eventually deciding on which forwarder to create in wallet platform and not sdk when it is calling wallet platform create address API [here](https://github.com/BitGo/BitGoJS/blob/303e0dd0e89e61a87d3c8edcb012012cf6b35f34/modules/sdk-core/src/bitgo/wallet/wallet.ts#L1215)

Reply:
Then I think we should somehow remove `defaultForwarderVersion`  probably in a follow-up ticket